packages/@ember/-internals/glimmer/lib/components/input.ts

```diff
@@ -112,9 +269,510 @@ import InternalComponent from './internal';
+              let arg = this[`_${argument}`];
+
+              if (arg) {
+                return `${superGetter.call(this)} ${this[`_${argument}`]}`;
+              } else {
+                return superGetter.call(this);
+              }
+            } else if (argument in this.args) {
+              return this[`_${argument}`];
+            } else {
+              return superGetter.call(this);
+            }
+          },
+        });
+      }
+    };
+
+    let deprecatedAttributeBindings: Array<
+      string | Parameters<typeof defineGetterForDeprecatedAttributeBinding>
+    > = [
+      // Component
+      'id',
+      'class',
+      ['class', 'classNames'],
+
+      // TextSupport
+      'autocapitalize',
+      'autocorrect',
+      'autofocus',
+      'disabled',
+      'form',
+      'maxlength',
+      'minlength',
+      'placeholder',
+      'readonly',
+      'required',
+      'selectionDirection',
+      'spellcheck',
+      'tabindex',
+      'title',
+
+      // TextField
+      'accept',
+      'autocomplete',
+      'autosave',
+      'dir',
+      'formaction',
+      'formenctype',
+      'formmethod',
+      'formnovalidate',
+      'formtarget',
+      'height',
+      'inputmode',
+      'lang',
+      'list',
+      'max',
+      'min',
+      'multiple',
+      'name',
+      'pattern',
+      'size',
+      'step',
+      'width',
+
+      // Checkbox
+      'indeterminate',
+    ];
+
+    deprecatedAttributeBindings.forEach((args) => {
+      if (Array.isArray(args)) {
+        defineGetterForDeprecatedAttributeBinding(...args);
+      } else {
+        defineGetterForDeprecatedAttributeBinding(args);
+      }
+    });
+  }
+
+  // Event callbacks
+  {
+    let defineGetterForDeprecatedEventCallback = (
+      eventName: string,
+      methodName: string = eventName,
+      virtualEvent?: string
+    ): void => {
+      assert(
+        `[BUG] There is already a getter for _${methodName} on Input`,
+        !Object.getOwnPropertyDescriptor(Input.prototype, `_${methodName}`)
+      );
+
+      let descriptor = Object.getOwnPropertyDescriptor(Input.prototype, methodName);
+
+      Object.defineProperty(Input.prototype, `_${methodName}`, {
+        get(this: Input): unknown {
+          return (event: Event): void => {
+            let value = this['valueFor'].call(this, event);
+
+            if (methodName in this.args) {
+              deprecate(
+                `Passing the \`@${methodName}\` argument to <Input> is deprecated. ` +
+                  `This would have overwritten the internal \`${methodName}\` method on ` +
+                  `the <Input> component and prevented it from functioning properly. ` +
+                  `Instead, please use the {{on}} modifier, i.e. \`<Input {{on "${eventName}" ...}} />\` ` +
+                  `instead of \`<Input @${methodName}={{...}} />\` or \`{{input ${methodName}=...}}\`.`,
+                true /* TODO !descriptor */,
```

Comment:
Can you open an issue with some guidance and then link back to it from here? Basically, we need _some_ path to enabling this deprecation...

Reply:
https://github.com/emberjs/ember.js/issues/19270